docs/getting_started/features/flexible_services.md

```diff
@@ -5,7 +5,281 @@ Flexible services, also called demand-responsibe services, are services that do
+    <p style="font-size:16px">
+    The following sample defines two different set of booking rules, the first one for trips that must be booked at least one day in advance (before 1PM) and no more than 14 days prior, and a second one for trips that can be booked at least 45 minutes prior to the trip and no more than 5 hours before.
+
+    </p>
+    !!! note ""
+        <p style="font-size:16px">
+        <a href="/documentation/schedule/reference/#booking_rulestxt"><b>booking_rules.txt</b></a> <br>
+        </p>
+
+        | booking_rule_id | booking_type | prior_notice_duration_min | prior_notice_duration_max | prior_notice_last_day | prior_notice_last_time | prior_notice_start_day | prior_notice_start_time | prior_notice_service_id | message                                                                                                                                            | pickup_message | drop_off_message | phone_number   | info_url             | booking_url             |
+        |-----------------|--------------|---------------------------|---------------------------|-----------------------|------------------------|------------------------|-------------------------|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|----------------|------------------|----------------|----------------------|-------------------------|
+        | route_br_1818   |            2 |                           |                           |                     1 |                  13:00 |                     14 |                    9:00 |                         | To request a ride, call 123-111-2233 before 1pm at least one business day ahead of your trip. You can book trips up to 14 business days in advace. |                |                  | (123)-111-2233 | flexservice.org/info | flexservice.org/booking |
+        | route_br_4545   |            1 |                        45 |                       300 |                       |                        |                        |                         |                         | To request a ride, use the official booking system in our website, trips must be booked at least 45 min in advance                                 |                |                  | (123)-111-2233 | flexservice.org/info | flexservice.org/booking |
+
+
+## Predefined routes with deviation
+
+Predefined routes with deviation can be used to model flexible services where vehicles can briefly deviate from a specific route to pick up users that booked a trip within a specific area along the route. This uses a combination of traditional stops (like a regular scheduled service) and zones using `locations.geojson`.
+
+**Pre-requirement**: 
+
+- [Base features](/getting_started/features/base)
+- [Booking rules feature](/getting_started/features/flexible_services/#booking-rules) recommended if service requires booking
```

Comment:
If, in the case of a service that requires booking, Booking rules _needs_ to be defined, I wouldn't use the term "recommended" here. I think we can remove it and simply have "Booking Rules feature if the service requires booking".

Reply:
Updated!